File: src/codes/Users_Manual/link/visualizer_phase.cpp
//~#include<autd3.hpp>
#include "autd3/link/visualizer.hpp"

//~int main() {
auto autd = autd3::ControllerBuilder({autd3::AUTD3(autd3::Vector3::Zero())})
                .open(autd3::link::Visualizer::builder());

autd3::Vector3 center = autd.geometry().center() + autd3::Vector3(0, 0, 150);
autd3::gain::Focus g(center);
autd.send(g);

autd3::link::PlotConfig config;
config.fname = "phase.png";
autd.link().plot_phase(config, autd3::Segment::S0, 0);
//~return 0; }

File: src/codes/Users_Manual/link/visualizer_phase.rs
# use autd3::prelude::*;
use autd3_link_visualizer::{Visualizer, PlotConfig};

use std::path::Path;

# #[allow(unused_variables)]
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
let mut autd = Controller::builder([AUTD3::new(Vector3::zeros())])
    .open(Visualizer::builder()).await?;

let center = autd.geometry().center() + Vector3::new(0., 0., 150.0 * mm);
let g = Focus::new(center);
autd.send(g).await?;

autd.link().plot_phase(
    PlotConfig {
        fname: Path::new("phase.png").into(),
        ..PlotConfig::default()
    },
    Segment::S0,
    0
)?;
# autd.close().await?;
# Ok(())
# }
